pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -530,19 +530,16 @@ func updateObjectsLabels(crLables map[string]string, objs []*unstructured.Unstru
 		// Label objects with version of the operator they were created by
 		labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
 		labels[names.PROMETHEUS_LABEL_KEY] = names.PROMETHEUS_LABEL_VALUE
-		err = updateObjectTemplateLabels(obj, labels, []string{names.PROMETHEUS_LABEL_KEY})
-		if err != nil {
-			return err
-		}
-
-		appLabelKeys := []string{names.COMPONENT_LABEL_KEY, names.PART_OF_LABEL_KEY, names.VERSION_LABEL_KEY, names.MANAGED_BY_LABEL_KEY}
+		labels[names.MANAGED_BY_LABEL_KEY] = names.MANAGED_BY_LABEL_DEFAULT_VALUE
 
+		appLabelKeys := []string{names.COMPONENT_LABEL_KEY, names.PART_OF_LABEL_KEY, names.VERSION_LABEL_KEY}
 		labels = updateLabelsFromCR(labels, crLables, appLabelKeys)
 		if err != nil {
 			return err
 		}
 		obj.SetLabels(labels)
 
+		appLabelKeys = append(appLabelKeys, names.PROMETHEUS_LABEL_KEY, names.MANAGED_BY_LABEL_KEY)
```

Comment:
now that there are 2 additional labels
maybe worth to declare an array of them, the ones that will be added by aren't updated from the CR
and then append this array instead appending individuals ?

Reply:
you are right, this deserves an individual string array